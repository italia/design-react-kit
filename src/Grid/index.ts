export { GridItem } from './GridItem';
export { GridItemText } from './GridItemText';
export { GridItemTextWrapper } from './GridItemTextWrapper';
export { GridList } from './GridList';
export { GridRow } from './GridRow';

export type { GridItemProps } from './GridItem';
export type { GridItemTextProps } from './GridItemText';
export type { GridItemTextWrapperProps } from './GridItemTextWrapper';
export type { GridListProps } from './GridList';
export type { GridRowProps } from './GridRow';
