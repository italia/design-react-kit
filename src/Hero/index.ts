export { Hero } from './Hero';
export { HeroBackground } from './HeroBackground';
export { HeroBody } from './HeroBody';
export { HeroButton } from './HeroButton';
export { HeroCategory } from './HeroCategory';
export { HeroTitle } from './HeroTitle';

export type { HeroProps } from './Hero';
export type { HeroBackgroundProps } from './HeroBackground';
export type { HeroBodyProps } from './HeroBody';
export type { HeroButtonProps } from './HeroButton';
export type { HeroCategoryProps } from './HeroCategory';
export type { HeroTitleProps } from './HeroTitle';
