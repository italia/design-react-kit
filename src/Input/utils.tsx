import classNames from 'classnames';
import { useState, useCallback } from 'react';
import type { CSSModule } from 'reactstrap';
import { mapToCssModules } from '../utils';
import type { InputProps } from './Input';

type ValidationProps = Pick<InputProps, 'valid' | 'invalid'>;
type TypeProps = Pick<InputProps, 'plaintext' | 'type'> & {
  staticInput?: boolean;
};
type FormControlProps = Pick<InputProps, 'addon' | 'normalized'> & TypeProps;

function getFormControlClassInternal({
  plaintext,
  staticInput,
  type = 'text',
  addon,
  normalized
}: FormControlProps) {
  const formControlClass = 'form-control';
  if (plaintext || staticInput || normalized) {
    return `${formControlClass}-plaintext`;
  }
  if (type === 'file') {
    return `${formControlClass}-file`;
  }
  if (['radio', 'checkbox'].indexOf(type) > -1) {
    if (addon) {
      return null;
    }
  }
  return formControlClass;
}

export function getFormControlClass(
  props: FormControlProps,
  cssModule?: CSSModule
) {
  return mapToCssModules(getFormControlClassInternal(props), cssModule);
}

export function getInfoTextControlClass(
  { valid, invalid }: ValidationProps,
  cssModule?: CSSModule
) {
  return mapToCssModules(
    classNames({
      'form-text': valid || invalid,
      'text-muted': !valid || invalid
    }),
    cssModule
  );
}

export function getTag({
  tag,
  plaintext,
  staticInput,
  type = 'text'
}: Pick<InputProps, 'tag'> & TypeProps) {
  if (tag) {
    return tag;
  }
  if (['radio', 'checkbox'].indexOf(type) > -1) {
    return 'input';
  }
  if (plaintext || staticInput) {
    return 'p';
  }
  return 'input';
}

type InputClassesParams = ValidationProps &
  Pick<
    InputProps,
    'bsSize' | 'placeholder' | 'value' | 'label' | 'infoText' | 'normalized'
  > & {
    isFocused: boolean;
    inputPassword?: boolean;
    formControlClass?: string;
    infoTextControlClass: string;
    originalWrapperClass: InputProps['wrapperClassName'];
  };

export function getClasses(
  className: string | undefined,
  {
    isFocused,
    valid,
    invalid,
    bsSize,
    placeholder,
    value,
    label,
    infoText,
    normalized,
    inputPassword,
    formControlClass,
    infoTextControlClass,
    originalWrapperClass
  }: InputClassesParams,
  cssModule?: CSSModule
) {
  const hasPlainCondition = placeholder || label || infoText;
  const baseCondition = hasPlainCondition && !normalized && !inputPassword;
  const passwordOnlyCondition =
    inputPassword && !hasPlainCondition && !normalized;
  const normalizedOnlyCondition =
    normalized && !hasPlainCondition && !inputPassword;

  const classes = mapToCssModules(
    classNames(
      className,
      {
        'is-invalid': invalid,
        'is-valid': valid,
        [`form-control-${bsSize}`]: bsSize
      },
      formControlClass
    ),
    cssModule
  );
  const wrapperClass = mapToCssModules(
    classNames(className, originalWrapperClass, 'form-group'),
    cssModule
  );
  const infoTextClass = mapToCssModules(
    classNames(
      {
        'valid-feedback': valid,
        'invalid-feedback': invalid
      },
      infoTextControlClass
    ),
    cssModule
  );

  const inputClasses = mapToCssModules(
    classNames(
      classes,
      !baseCondition && {
        // we can model here only if stylings
        'form-control-plaintext': normalizedOnlyCondition,
        'form-control': passwordOnlyCondition,
        'input-password': passwordOnlyCondition,
        'focus--mouse': passwordOnlyCondition || normalizedOnlyCondition
      }
    ),
    cssModule
  );

  const activeClass = mapToCssModules(
    classNames({
      active: isFocused || placeholder || value
    }),
    cssModule
  );

  return {
    wrapperClass,
    inputClasses,
    activeClass,
    infoTextClass
  };
}

export function useFocus<T extends HTMLInputElement | HTMLTextAreaElement>({
  onFocus,
  onBlur
}: {
  onFocus: React.FocusEventHandler<T> | undefined;
  onBlur: React.FocusEventHandler<T> | undefined;
}) {
  const [isFocused, setFocus] = useState(false);

  const toggleFocusLabel = useCallback(
    (e: React.FocusEvent<T>) => {
      setFocus(true);
      onFocus?.(e);
    },
    [onFocus]
  );

  const toggleBlurLabel = useCallback(
    (e: React.FocusEvent<T>) => {
      if (e.target.value === '') {
        setFocus(!isFocused);
      }
      onBlur?.(e);
    },
    [isFocused, onBlur]
  );

  return { toggleFocusLabel, toggleBlurLabel, isFocused };
}
