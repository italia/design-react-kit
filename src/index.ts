export {
  BreadcrumbItem,
  ButtonGroup,
  ButtonToolbar,
  CardColumns,
  CardDeck,
  CardFooter,
  CardGroup,
  CardHeader,
  CardImg,
  CardImgOverlay,
  CardLink,
  CardSubtitle,
  Col,
  Container,
  Fade,
  Form,
  FormFeedback,
  FormGroup,
  FormText,
  InputGroup,
  InputGroupText,
  Label,
  ListGroup,
  ListGroupItem,
  ListGroupItemHeading,
  ListGroupItemText,
  Media,
  NavItem,
  NavLink,
  Navbar,
  NavbarBrand,
  NavbarToggler,
  PaginationItem as PagerItem,
  PaginationLink as PagerLink,
  Popover,
  PopoverBody,
  PopoverHeader,
  Row,
  TabContent,
  TabPane,
  Table,
  Tooltip,
  UncontrolledAlert,
  UncontrolledCollapse,
  UncontrolledTooltip,
  Util
} from 'reactstrap';

export { Accordion } from './Accordion/Accordion';
export { AccordionBody } from './Accordion/AccordionBody';
export { AccordionHeader } from './Accordion/AccordionHeader';
export { AccordionItem } from './Accordion/AccordionItem';
export { Alert } from './Alert/Alert';
export { AvatarContainer } from './Avatar/AvatarContainer';
export { AvatarExtraText } from './Avatar/AvatarExtraText';
export { AvatarGroupContainer } from './Avatar/AvatarGroupContainer';
export { AvatarIcon } from './Avatar/AvatarIcon';
export { AvatarPresence } from './Avatar/AvatarPresence';
export { AvatarStatus } from './Avatar/AvatarStatus';
export { AvatarWrapper } from './Avatar/AvatarWrapper';
export { BackToTop } from './BackToTop/BackToTop';
export { Badge } from './Badge/Badge';
export { BottomNav } from './BottomNav/BottomNav';
export { BottomNavItem } from './BottomNav/BottomNavItem';
export { Breadcrumb } from './Breadcrumb/Breadcrumb';
export { Button } from './Button/Button';
export { Callout } from './Callout/Callout';
export { CalloutMoreFooter } from './Callout/CalloutMoreFooter';
export { CalloutText } from './Callout/CalloutText';
export { CalloutTitle } from './Callout/CalloutTitle';
export { Card } from './Card/Card';
export { CardBody } from './Card/CardBody';
export { CardCategory } from './Card/CardCategory';
export { CardFooterCTA } from './Card/CardFooterCTA';
export { CardReadMore } from './Card/CardReadMore';
export { CardSignature } from './Card/CardSignature';
export { CardTag } from './Card/CardTag';
export { CardTagsHeader } from './Card/CardTagsHeader';
export { CardText } from './Card/CardText';
export { CardTitle } from './Card/CardTitle';
export { Chip } from './Chips/Chip';
export { ChipLabel } from './Chips/ChipLabel';
export { Collapse } from './Collapse/Collapse';
export { Dimmer } from './Dimmer/Dimmer';
export { DimmerButtons } from './Dimmer/DimmerButtons';
export { Dropdown } from './Dropdown/Dropdown';
export { DropdownMenu } from './Dropdown/DropdownMenu';
export { DropdownToggle } from './Dropdown/DropdownToggle';
export { FontLoader, useFontLoader } from './FontLoader/FontLoader';
export { Forward } from './Forward/Forward';
export { GoBack } from './GoBack/GoBack';
export { GridItem, GridItemText, GridItemTextWrapper, GridList, GridRow } from './Grid/index';
export { Header } from './Header/Header';
export { HeaderBrand } from './Header/HeaderBrand';
export { HeaderContent } from './Header/HeaderContent';
export { HeaderLinkZone } from './Header/HeaderLinkZone';
export { HeaderRightZone } from './Header/HeaderRightZone';
export { HeaderSearch } from './Header/HeaderSearch';
export { HeaderSocialsZone } from './Header/HeaderSocialsZone';
export { HeaderToggler } from './Header/HeaderToggler';
export { Headers } from './Header/Headers';
export { Hero, HeroBackground, HeroBody, HeroButton, HeroCategory, HeroTitle } from './Hero/index';
export { Icon, clearIconCache, iconsList as icons, preloadIcons } from './Icon/Icon';
export { Input } from './Input/Input';
export { InputContainer } from './Input/InputContainer';
export { TextArea } from './Input/TextArea';
export { LinkList } from './LinkList/LinkList';
export { LinkListItem } from './LinkList/LinkListItem';
export { MegamenuFooter } from './Megamenu/MegamenuFooter';
export { MegamenuHighlightColumn } from './Megamenu/MegamenuHighlightColumn';
export { MegamenuItem } from './Megamenu/MegamenuItem';
export { Modal } from './Modal/Modal';
export { ModalBody } from './Modal/ModalBody';
export { ModalFooter } from './Modal/ModalFooter';
export { ModalHeader } from './Modal/ModalHeader';
export { Nav } from './Nav/Nav';
export { useNavScroll } from './NavScroll/index';
export { NotificationManager, createNotification, notify } from './Notification/index';
export { Pager } from './Pager/Pager';
export { Progress } from './Progress/Progress';
export { Rating } from './Rating/Rating';
export { ResponsiveImage } from './ResponsiveImage/ResponsiveImage';
export { Section } from './Section/Section';
export { Select } from './Select/Select';
export { Sidebar } from './Sidebar/Sidebar';
export { Skiplink } from './Skiplink/Skiplink';
export { SkiplinkItem } from './Skiplink/SkiplinkItem';
export { Spinner } from './Spinner/Spinner';
export { StepperContainer } from './Stepper/StepperContainer';
export { StepperContent } from './Stepper/StepperContent';
export { StepperDots } from './Stepper/StepperDots';
export { StepperHeader } from './Stepper/StepperHeader';
export { StepperHeaderElement } from './Stepper/StepperHeaderElement';
export { StepperNav } from './Stepper/StepperNav';
export { ThumbNav } from './ThumbNav/ThumbNav';
export { ThumbNavItem } from './ThumbNav/ThumbNavItem';
export { TimelinePin } from './Timeline/TimelinePin';
export { Timeline } from './Timeline/TimelineWrapper';
export { Toggle } from './Toggle/Toggle';
export { Toolbar } from './Toolbar/Toolbar';
export { ToolbarDividerItem } from './Toolbar/ToolbarDividerItem';
export { ToolbarItem } from './Toolbar/ToolbarItem';

// Types
export type { AccordionProps } from './Accordion/Accordion';
export type { AccordionBodyProps } from './Accordion/AccordionBody';
export type { AccordionHeaderProps } from './Accordion/AccordionHeader';
export type { AlertProps } from './Alert/Alert';
export type { AvatarContainerProps } from './Avatar/AvatarContainer';
export type { AvatarExtraTextProps } from './Avatar/AvatarExtraText';
export type { AvatarGroupContainerProps } from './Avatar/AvatarGroupContainer';
export type { AvatarIconProps } from './Avatar/AvatarIcon';
export type { AvatarPresenceProps } from './Avatar/AvatarPresence';
export type { AvatarStatusProps } from './Avatar/AvatarStatus';
export type { AvatarWrapperProps } from './Avatar/AvatarWrapper';
export type { BadgeProps } from './Badge/Badge';
export type { BottomNavProps } from './BottomNav/BottomNav';
export type { BottomNavItemProps } from './BottomNav/BottomNavItem';
export type { BreadcrumbProps } from './Breadcrumb/Breadcrumb';
export type { ButtonProps } from './Button/Button';
export type { CalloutProps } from './Callout/Callout';
export type { CalloutTextProps } from './Callout/CalloutText';
export type { CalloutTitleProps } from './Callout/CalloutTitle';
export type { CardProps } from './Card/Card';
export type { CardBodyProps } from './Card/CardBody';
export type { CardCategoryProps } from './Card/CardCategory';
export type { CardFooterCTAProps } from './Card/CardFooterCTA';
export type { CardReadMoreProps } from './Card/CardReadMore';
export type { CardSignatureProps } from './Card/CardSignature';
export type { CardTagProps } from './Card/CardTag';
export type { CardTagsHeaderProps } from './Card/CardTagsHeader';
export type { CardTextProps } from './Card/CardText';
export type { CardTitleProps } from './Card/CardTitle';
export type { ChipProps } from './Chips/Chip';
export type { ChipLabelProps } from './Chips/ChipLabel';
export type { DimmerProps } from './Dimmer/Dimmer';
export type { DimmerButtonsProps } from './Dimmer/DimmerButtons';
export type { DropdownProps } from './Dropdown/Dropdown';
export type { DropdownMenuProps } from './Dropdown/DropdownMenu';
export type { DropdownToggleProps } from './Dropdown/DropdownToggle';
export type { FontLoaderProps } from './FontLoader/FontLoader';
export type { ForwardProps } from './Forward/Forward';
export type {
  GridItemProps,
  GridItemTextProps,
  GridItemTextWrapperProps,
  GridListProps,
  GridRowProps
} from './Grid/index';
export type { HeaderProps } from './Header/Header';
export type { HeaderBrandProps } from './Header/HeaderBrand';
export type { HeaderContentProps } from './Header/HeaderContent';
export type { HeaderLinkZoneProps } from './Header/HeaderLinkZone';
export type { HeaderRightZoneProps } from './Header/HeaderRightZone';
export type { HeaderSearchProps } from './Header/HeaderSearch';
export type { HeaderSocialsZoneProps } from './Header/HeaderSocialsZone';
export type { HeaderTogglerProps } from './Header/HeaderToggler';
export type { HeadersProps } from './Header/Headers';
export type {
  HeroBackgroundProps,
  HeroBodyProps,
  HeroButtonProps,
  HeroCategoryProps,
  HeroProps,
  HeroTitleProps
} from './Hero/index';
export type { IconName, IconProps } from './Icon/Icon';
export type { InputProps } from './Input/Input';
export type { InputContainerProps } from './Input/InputContainer';
export type { TextAreaProps } from './Input/TextArea';
export type { LinkListProps } from './LinkList/LinkList';
export type { LinkListItemProps } from './LinkList/LinkListItem';
export type { MegamenuFooterProps } from './Megamenu/MegamenuFooter';
export type { MegamenuHCProps } from './Megamenu/MegamenuHighlightColumn';
export type { MegamenuItemProps } from './Megamenu/MegamenuItem';
export type { ModalProps } from './Modal/Modal';
export type { ModalBodyProps } from './Modal/ModalBody';
export type { ModalFooterProps } from './Modal/ModalFooter';
export type { ModalHeaderProps } from './Modal/ModalHeader';
export type { NavProps } from './Nav/Nav';
export type { ChangesType, RegisterOptions, useNavScrollArgs, useNavScrollResult } from './NavScroll';
export type { NotificationId, NotificationManagerProps, NotificationOptions, NotificationProps } from './Notification';
export type { PagerProps } from './Pager/Pager';
export type { ProgressProps } from './Progress/Progress';
export type { RatingProps } from './Rating/Rating';
export type { ResponsiveImageProps } from './ResponsiveImage/ResponsiveImage';
export type { SectionProps } from './Section/Section';
export type { SelectProps } from './Select/Select';
export type { SidebarProps } from './Sidebar/Sidebar';
export type { SkiplinkProps } from './Skiplink/Skiplink';
export type { SkiplinkItemProps } from './Skiplink/SkiplinkItem';
export type { SpinnerProps } from './Spinner/Spinner';
export type { StepperContainerProps } from './Stepper/StepperContainer';
export type { StepperContentProps } from './Stepper/StepperContent';
export type { StepperDotsProps } from './Stepper/StepperDots';
export type { StepperHeaderProps } from './Stepper/StepperHeader';
export type { StepperHeaderElementProps } from './Stepper/StepperHeaderElement';
export type { StepperNavProps } from './Stepper/StepperNav';
export type { ThumbNavProps } from './ThumbNav/ThumbNav';
export type { ThumbNavItemProps } from './ThumbNav/ThumbNavItem';
export type { TimelinePinProps } from './Timeline/TimelinePin';
export type { TimelineProps } from './Timeline/TimelineWrapper';
export type { ToggleProps } from './Toggle/Toggle';
export type { ToolbarProps } from './Toolbar/Toolbar';
export type { ToolbarItemBadge, ToolbarItemProps } from './Toolbar/ToolbarItem';

export type {
  BreadcrumbItemProps,
  ButtonDropdownProps,
  ButtonGroupProps,
  ButtonToolbarProps,
  CardColumnsProps,
  CardDeckProps,
  CardFooterProps,
  CardGroupProps,
  CardHeaderProps,
  CardImgOverlayProps,
  CardImgProps,
  CardLinkProps,
  CardSubtitleProps,
  ColProps,
  ContainerProps,
  FadeProps,
  FormFeedbackProps,
  FormGroupProps,
  FormProps,
  FormTextProps,
  InputGroupProps,
  InputGroupTextProps,
  LabelProps,
  ListGroupItemHeadingProps,
  ListGroupItemProps,
  ListGroupItemTextProps,
  ListGroupProps,
  MediaProps,
  NavItemProps,
  NavLinkProps,
  NavbarBrandProps,
  NavbarProps,
  NavbarTogglerProps,
  PaginationItem as PagerItemProps,
  PaginationLink as PagerLinkProps,
  PopoverBodyProps,
  PopoverHeaderProps,
  PopoverProps,
  RowProps,
  TabContentProps,
  TabPaneProps,
  TableProps,
  TooltipProps,
  UncontrolledAlertProps,
  UncontrolledCollapseProps,
  UncontrolledTooltipProps
} from 'reactstrap';
