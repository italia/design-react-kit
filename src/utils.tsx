export const noop = () => {};

export const logError = (message: string) => {
  if (typeof console !== 'undefined') {
    if (console.error) {
      console.error(message);
    } else {
      console.log(message);
    }
  }
};

// @internal Used for testing purposes only
export const flushMessageCache = () => {
  for (const key in messages) {
    delete messages[key];
  }
};

const messages: Record<string, 1> = {};
export const notifyDeprecation = (
  message: string,
  options: { once: boolean } = { once: true }
) => {
  if (!options.once) {
    logError(message);
  } else {
    if (!messages[message]) {
      logError(message);
      messages[message] = 1;
    }
  }
};

export function mapToCssModules(
  className: string | null,
  cssModules?: Record<string, string>
) {
  let finalClassNames = className == null ? '' : className;
  if (!cssModules) {
    return finalClassNames;
  }
  return finalClassNames
    .split(' ')
    .map((klass) => cssModules[klass] ?? klass)
    .join(' ');
}

type UnknownObject = Record<string, unknown>;

export function pick<T extends UnknownObject>(
  obj: T,
  keys: keyof T | Array<keyof T>
): Partial<T> {
  const keysArray = Array.isArray(keys) ? keys : [keys];
  const newObj: Partial<T> = {};
  for (const key of keysArray) {
    if (key in obj) {
      newObj[key] = obj[key];
    }
  }
  return newObj;
}

export function omit<T extends UnknownObject>(
  obj: T,
  keys: keyof T | Array<keyof T>
): Partial<T> {
  const keysLookup = new Set(Array.isArray(keys) ? keys : [keys]);

  const newObj: Partial<T> = {};
  for (const key in obj) {
    if (!keysLookup.has(key)) {
      newObj[key] = obj[key];
    }
  }
  return newObj;
}
