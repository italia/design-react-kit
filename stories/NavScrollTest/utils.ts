export function getStorybookExtraStyles() {
  return insideIframe() && insideDocsPage()
    ? {
        maxHeight: 600,
        overflowY: 'scroll' as const
      }
    : {};
}

function insideDocsPage() {
  return document.querySelector('.innerZoomElementWrapper');
}

function insideIframe() {
  try {
    return window.self !== window.top;
  } catch (e) {
    return true;
  }
}
