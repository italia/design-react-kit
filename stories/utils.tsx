export const requireContextPolyfill = () => {
  if (process.env.NODE_ENV === 'test') {
    const fs = require('fs');
    const path = require('path');

    require.context = (base = '.', scanSubDirectories = false, regularExpression = /\.js$/) => {
      const files = {};

      function readDirectory(directory) {
        fs.readdirSync(directory).forEach((file) => {
          const fullPath = path.resolve(directory, file);

          if (fs.statSync(fullPath).isDirectory()) {
            if (scanSubDirectories) readDirectory(fullPath);

            return;
          }

          if (!regularExpression.test(fullPath)) return;

          files[fullPath] = true;
        });
      }

      readDirectory(path.resolve(__dirname, base));

      function Module(file) {
        return require(file);
      }

      Module.keys = () => Object.keys(files);

      return Module;
    };

    return require.context;
  }
};

export const envIs = (env) => {
  return process.env.NODE_ENV === env;
};
