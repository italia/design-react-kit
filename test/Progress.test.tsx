import "@testing-library/jest-dom";
import { render, screen } from '@testing-library/react';
import React from 'react';

import { Progress } from '../src';
import { logError } from '../src/utils';

jest.mock('../src/utils', () => ({
  logError: jest.fn()
}));

beforeEach(() => {
  (logError as jest.Mock).mockRestore();
});

// Tests for breaking changes here
test('Should support old className behaviour with special flag', () => {
  const { container } = render(<Progress className='myClass' wrapperClassName={true} value={1} />);
  expect(container.firstChild).toHaveClass('myClass');
});

test('Should not apply className to the wrapper', () => {
  const { container } = render(<Progress className='myClass' value={1} />);
  expect(container.firstChild).not.toHaveClass('myClass');
});

test('Should apply wrapperClassName to the wrapper', () => {
  const { container } = render(<Progress wrapperClassName='myClass' value={1} />);
  expect(container.firstChild).toHaveClass('myClass');
});

// Write more proper tests here
test('Should log an error when value is not a number', () => {
  const { container } = render(<Progress wrapperClassName='myClass' value={'a'} />);
  expect(container.firstChild).toHaveClass('myClass');
  expect(logError).toHaveBeenCalledWith('The passed "value" is not a valid number. You passed "a"');
});

test('Should be ok with a numeric string as for value', () => {
  const { container } = render(<Progress wrapperClassName='myClass' value={'1'} />);
  expect(container.firstChild).toHaveClass('myClass');
});

test('should have a testId for resilient UI changes', () => {
  render(<Progress testId='test-id-progress' />);

  expect(screen.getByTestId('test-id-progress')).toBeTruthy();
});
