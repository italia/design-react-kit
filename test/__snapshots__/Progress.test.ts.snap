// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stories Snapshots Documentazione/Componenti/Progress _EsempiInterattivi 1`] = `
<div>
  <div>
    <div
      class="progress-bar-wrapper"
    >
      <div
        class="progress-bar-label"
      >
        <span
          class="visually-hidden"
        >
          progresso
        </span>
        25%
      </div>
      <div
        class="progress"
      >
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="25"
          class="progress-bar"
          role="progressbar"
          style="width: 25%;"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Stories Snapshots Documentazione/Componenti/Progress BottoneConProgressBar 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="col-12 col-sm-6"
      >
        <p>
          <strong>
            Bottone primario
          </strong>
        </p>
        <button
          aria-disabled="true"
          class="btn-progress btn btn-primary disabled"
          disabled=""
        >
          Label bottone 
          <svg
            aria-hidden="true"
            class="icon icon-light"
            enable-background="new 0 0 24 24"
            role="img"
            viewBox="0 0 24 24"
            xml:space="preserve"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              icon
            </title>
            <path
              d="M12 2c-4.9 0-9.1 3.6-9.9 8.4s2.1 9.5 6.7 11.1c.5.1.7-.2.7-.5v-1.7c-2.8.6-3.4-1.3-3.4-1.3-.1-.6-.6-1.2-1.1-1.5-.9-.6.1-.6.1-.6.6.1 1.2.5 1.5 1 .6 1 1.9 1.4 2.9.8 0-.5.3-1 .6-1.3-2.1-.2-4.5-1.1-4.5-4.9 0-1 .3-2 1-2.7-.3-.9-.2-1.8.1-2.6 0 0 .8-.3 2.8 1 1.6-.4 3.4-.4 5 0 1.9-1.3 2.7-1 2.7-1 .4.8.4 1.8.1 2.6.7.7 1 1.7 1 2.7 0 3.8-2.3 4.7-4.6 4.9.5.5.7 1.2.7 1.9v2.8c0 .4.2.6.7.5 4.7-1.5 7.5-6.2 6.7-11.1S16.9 2 12 2z"
            />
          </svg>
          <span>
            <div
              class="progress"
            >
              <div
                class="progress-bar-label"
              >
                <span
                  class="visually-hidden"
                />
              </div>
              <div
                aria-valuemax="100"
                aria-valuemin="0"
                aria-valuenow="50"
                class="progress-bar"
                role="progressbar"
                style="width: 50%;"
              />
            </div>
          </span>
        </button>
      </div>
      <div
        class="col-12 col-sm-6"
      >
        <p>
          <strong>
            Bottone secondario
          </strong>
        </p>
        <button
          aria-disabled="true"
          class="btn-progress btn btn-secondary disabled"
          disabled=""
        >
          Label bottone 
          <svg
            aria-hidden="true"
            class="icon icon-light"
            enable-background="new 0 0 24 24"
            role="img"
            viewBox="0 0 24 24"
            xml:space="preserve"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              icon
            </title>
            <path
              d="M12 2c-4.9 0-9.1 3.6-9.9 8.4s2.1 9.5 6.7 11.1c.5.1.7-.2.7-.5v-1.7c-2.8.6-3.4-1.3-3.4-1.3-.1-.6-.6-1.2-1.1-1.5-.9-.6.1-.6.1-.6.6.1 1.2.5 1.5 1 .6 1 1.9 1.4 2.9.8 0-.5.3-1 .6-1.3-2.1-.2-4.5-1.1-4.5-4.9 0-1 .3-2 1-2.7-.3-.9-.2-1.8.1-2.6 0 0 .8-.3 2.8 1 1.6-.4 3.4-.4 5 0 1.9-1.3 2.7-1 2.7-1 .4.8.4 1.8.1 2.6.7.7 1 1.7 1 2.7 0 3.8-2.3 4.7-4.6 4.9.5.5.7 1.2.7 1.9v2.8c0 .4.2.6.7.5 4.7-1.5 7.5-6.2 6.7-11.1S16.9 2 12 2z"
            />
          </svg>
          <span>
            <div
              class="progress"
            >
              <div
                class="progress-bar-label"
              >
                <span
                  class="visually-hidden"
                />
              </div>
              <div
                aria-valuemax="100"
                aria-valuemin="0"
                aria-valuenow="50"
                class="progress-bar"
                role="progressbar"
                style="width: 50%;"
              />
            </div>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Stories Snapshots Documentazione/Componenti/Progress Colori 1`] = `
<div>
  <div>
    <div
      class="progress progress-color"
    >
      <div
        class="progress-bar-label"
      >
        <span
          class="visually-hidden"
        />
      </div>
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="25"
        class="progress-bar bg-success"
        role="progressbar"
        style="width: 25%;"
      />
    </div>
    <div
      class="progress progress-color"
    >
      <div
        class="progress-bar-label"
      >
        <span
          class="visually-hidden"
        />
      </div>
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="33"
        class="progress-bar bg-info"
        role="progressbar"
        style="width: 33%;"
      />
    </div>
    <div
      class="progress progress-color"
    >
      <div
        class="progress-bar-label"
      >
        <span
          class="visually-hidden"
        />
      </div>
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="50"
        class="progress-bar bg-warning"
        role="progressbar"
        style="width: 50%;"
      />
    </div>
    <div
      class="progress progress-color"
    >
      <div
        class="progress-bar-label"
      >
        <span
          class="visually-hidden"
        />
      </div>
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="66"
        class="progress-bar bg-danger"
        role="progressbar"
        style="width: 66%;"
      />
    </div>
  </div>
</div>
`;

exports[`Stories Snapshots Documentazione/Componenti/Progress Esempio 1`] = `
<div>
  <div
    class="progress"
  >
    <div
      class="progress-bar-label"
    >
      <span
        class="visually-hidden"
      />
    </div>
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="50"
      class="progress-bar"
      role="progressbar"
      style="width: 50%;"
    />
  </div>
</div>
`;

exports[`Stories Snapshots Documentazione/Componenti/Progress Etichette 1`] = `
<div>
  <div
    class="progress-bar-wrapper"
  >
    <div
      class="progress-bar-label"
    >
      <span
        class="visually-hidden"
      >
        progresso
      </span>
      35%
    </div>
    <div
      class="progress"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="35"
        class="progress-bar"
        role="progressbar"
        style="width: 35%;"
      />
    </div>
  </div>
</div>
`;

exports[`Stories Snapshots Documentazione/Componenti/Progress ProgressoIndeterminato 1`] = `
<div>
  <div
    class="progress progress-indeterminate"
  >
    <div
      class="progress-bar-label"
    >
      <span
        class="visually-hidden"
      >
        In elaborazione...
      </span>
    </div>
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="NaN"
      class="progress-bar"
      role="progressbar"
    />
  </div>
</div>
`;
